pages/dashboard/[guildId]/[tokenId].tsx

```diff
@@ -104,15 +115,33 @@ export const getServerSideProps: GetServerSideProps = async ({
 }) => {
   await setupDb();
 
-  const { guildId } = query;
+  const { guildId, tokenId } = query;
 
-  if (!guildId || typeof guildId !== "string") {
+  if (
+    !guildId ||
+    typeof guildId !== "string" ||
+    !tokenId ||
+    typeof tokenId !== "string"
+  ) {
     if (res) res.statusCode = 400;
     return {
       props: {
         configs: [],
         guildId: "",
-        error: "Missing or invalid guild ID.",
+        error: "Missing or invalid guild or token ID.",
+      },
+    };
+  }
+  // ✅ Step: validate token
+  const isValidToken = await validateToken(guildId, tokenId);
```

Comment:
You need to create a new valideDashboardToken because this function is dedicated to the analytics.
You can rename this validateToken function to "validateAnalyticsToken" so that's it'll be clearer